DatadogSessionReplay/Tests/Recorder/RecordingCoordinatorTests.swift

```diff
@@ -33,38 +35,53 @@ class RecordingCoordinatorTests: XCTestCase {
 
     // MARK: Configuration Tests
 
-    func test_itDoesNotStartScheduler_afterInitializing() {
+    func test_itDoesNotStart_afterInitializing() {
         prepareRecordingCoordinator(sampler: Sampler(samplingRate: .mockRandom(min: 0, max: 100)))
-        XCTAssertFalse(scheduler.isRunning)
+
+        recordingTriggerMock.triggerCallback?()
+
+        XCTAssertEqual(recordingMock.captureNextRecordCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.startWatchingTriggersCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.stopWatchingTriggersCallsCount, 0)
+
+        rumContextObserver.notify(rumContext: .mockRandom())
+
         XCTAssertEqual(recordingMock.captureNextRecordCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.startWatchingTriggersCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.stopWatchingTriggersCallsCount, 1)
     }
 
-    func test_whenNotSampled_itStopsScheduler_andShouldNotRecord() {
+    func test_whenNotSampledAndTriggered_itShouldNotRecord() {
         // Given
         prepareRecordingCoordinator(sampler: .mockRejectAll())
+        recordingCoordinator?.startRecording()
 
         // When
         rumContextObserver.notify(rumContext: .mockRandom())
+        recordingTriggerMock.triggerCallback?()
 
         // Then
-        XCTAssertFalse(scheduler.isRunning)
         XCTAssertEqual(try core.context.baggages["sr_has_replay"]?.decode(), false)
         XCTAssertEqual(recordingMock.captureNextRecordCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.startWatchingTriggersCallsCount, 0)
+        XCTAssertEqual(recordingTriggerMock.stopWatchingTriggersCallsCount, 2)
```

Comment:
question/ Why `2`? I'd expect it to be stopped only once. Asking because `stopWatchingTriggers()` implies unswizzling which manages a `lock`. If this is vastly overcalled, may cause perf penalty.

Reply:
It's because `rumContext` change triggers recording `evaluateRecordingConditions` which starts/stops based on some criteria.

There's a simple way to improve this by knowing if it's recording or not. I'll add that change.